### PR TITLE
start highlighting squares

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: client/src/board.rs [hunk 1]
```diff
@@ -315,6 +315,22 @@ pub fn Board(cx: Scope<BoardProps>) -> Element {
...
+                let (top_left, z_index) = get_positions(cx, &pos, mouse_down_state, dragging_point_state);
+                dbg!(pos);
+                dbg!(top_left);
+                dbg!(z_index);
+
+                rsx! {
+                    div {
+                        class: "squares",
+                        style: "left: {top_left.x}px; top: {top_left.y}px; z-index: {z_index}",
+                        width: "{cx.props.size / 8}",
```
Comment (DanielAlbl) [inline client/src/board.rs:329]: I think this needs `px`.

File: client/src/board.rs [hunk 2]
```diff
@@ -315,6 +315,22 @@ pub fn Board(cx: Scope<BoardProps>) -> Element {
                     }
                 }
             }),
+            (0..64).map(|i| {
+                let pos = Position::new(i % 8, i / 8);
```
[the remark this answers is not in the document]
Comment (Dillon-Roller) [reply client/src/board.rs:315]: I dont know if there would be a use for it.. Like grabbing all pieces starting from 0 up to i of the chess board?

File: client/src/board.rs [hunk 3]
```diff
@@ -224,7 +228,7 @@ fn handle_on_mouse_move_event(
     mouse_down_state: &UseState<Option<MouseClick>>,
     dragging_point_state: &UseState<Option<ClientPoint>>,
 ) {
-    if mouse_down_state.get().is_some() {
+    if mouse_down_state.is_some() {
```
Comment (DanielAlbl) [inline client/src/board.rs:231]: Holy shit

File: client/src/board.rs [hunk 4]
```diff
@@ -291,28 +298,67 @@ pub fn Board(cx: Scope<BoardProps>) -> Element {
...
                 rsx! {
+
                     img {
                         src: "{piece_img}",
                         class: "images",
                         style: "left: {top_left.x}px; top: {top_left.y}px; z-index: {z_index}",
                         width: "{cx.props.size / 8}",
                         height: "{cx.props.size / 8}",
                     }
+                    div {
```
Comment (DanielAlbl) [inline client/src/board.rs:356]: Do we need this?

File: client/src/board.rs [hunk 5]
```diff
@@ -291,28 +298,67 @@ pub fn Board(cx: Scope<BoardProps>) -> Element {
...
+            }
+            // highlight from-to squares
+            if let Some(mv) = last_move.get() {
+                rsx! {
+                    mv.get_positions().into_iter().map(|pos| {
+                        let (top_left, _) = get_positions(cx, &pos, mouse_down_state, dragging_point_state);
+                        rsx! {
+                            div {
+                                class: {
+                                    if mv.from == pos || mv.to == pos {
```
Comment (DanielAlbl) [inline client/src/board.rs:326]: Will this always be true?

**Review by DanielAlbl — COMMENTED** [review]
Nice

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**